clang/lib/AST/ASTContext.cpp

```diff
@@ -14386,7 +14406,7 @@ QualType ASTContext::getCommonSugaredType(QualType X, QualType Y,
       SY = Ys.pop_back_val();
     }
   }
-  if (Unqualified)
+  if (KeepCommonQualifiers)
```

Comment:
I apologize if I am being dense here but having if `KeepCommonQualifiers` and then going on to calling `removeCommonQualifiers` feels confusing.

Reply:
Yeah, `removeCommonQualifiers` takes two parameters by reference, and removes their common qualifiers modifying them in place...
BUT, then it returns the common qualifiers, which is what we are interested, the rest is discarded.